pkg/config/process.go

```diff
@@ -70,3 +94,48 @@ func setupProcesses(config Config) {
 	)
 	procBindEnvAndSetDefault(config, "process_config.process_discovery.interval", 4*time.Hour)
 }
+
+var displayProcConfigEnabledWarningOnce sync.Once
+
+// displayProcConfigEnabledDeprecationWarning displays a deprecation warning for process_config.enabled only once.
+// For testing purposes, it returns a bool describing if the message was displayed or not
+func displayProcConfigEnabledDeprecationWarning() (displayed bool) {
+	displayProcConfigEnabledWarningOnce.Do(func() {
+		log.Debug("process_config.enabled is deprecated, use process_config.container_collection.enabled" +
+			" and process_config.process_collection.enabled instead")
+		displayed = true
+	})
+	return
+}
+
+// GetContainerCollectionEnabled retrieves the value of process_config.container_collection.enabled.
+// If process_config.enabled is set, we display a deprecation warning and use that value instead.
+func GetContainerCollectionEnabled(config Config) bool {
+	if config.IsSet("process_config.enabled") {
+		_ = displayProcConfigEnabledDeprecationWarning()
+
+		procConfigEnabled := strings.ToLower(config.GetString("process_config.enabled"))
+		if procConfigEnabled == "disabled" {
+			return false
+		}
+		result, _ := strconv.ParseBool(procConfigEnabled)
+		return !result
+	}
+	return config.GetBool("process_config.container_collection.enabled")
+}
+
+// GetProcessCollectionEnabled retrieves the value of process_config.process_collection.enabled.
+// If process_config.enabled is set, we display a deprecation warning and use that value instead.
+func GetProcessCollectionEnabled(config Config) bool {
+	if config.IsSet("process_config.enabled") {
+		_ = displayProcConfigEnabledDeprecationWarning()
+
+		procConfigEnabled := strings.ToLower(config.GetString("process_config.enabled"))
+		if procConfigEnabled == "disabled" {
+			return false
+		}
+		enabled, _ := strconv.ParseBool(procConfigEnabled)
+		return enabled
```

Comment:
I would suggest setting `process_config.process_collection.enabled` rather than requiring folks to call this helper function, for the case when they are using the old configuration. It simplifies and decouples other code, in that they only have to check one value and can just call `config.GetBool("process_config.process_collection.enabled")` to know if this is enabled. 

I used this approach when transitioning the NPM enable flag.

Reply:
Hey Bryce!

The goal with this config refactor is to reduce our reliance on custom config parsing in `pkg/process/config`. I see the value in not requiring folks to call a helper function, but I worry about adding a bunch of custom config handling logic into `pkg/process/config.LoadYamlConfig()` goes against the spirit of the RFC this was based off of. Let me know what you think about this, as I'm a bit conflicted on this.

Reply:
So you need this code either way, the difference is this:
Your current implementation does the parsing each time it is called. 
vs
My suggestion would do the parsing once during initial config load, and other users would use the config API they are already using for everything else.

For example, here is how we handle this in system-probe: https://github.com/DataDog/datadog-agent/blob/2c4b2fc06a4e5a3d2bd9a7f7fa8bc51f585b4175/cmd/system-probe/config/config.go#L225-L228
And other places can do this: https://github.com/DataDog/datadog-agent/blob/2c4b2fc06a4e5a3d2bd9a7f7fa8bc51f585b4175/pkg/status/status.go#L75

This is just a suggestion, so do what you think is best. 🙇

Reply:
That makes sense, and I think it might remove a lot of boilerplate code that is required to implement this. I'm going to try and implement it this way and see which one my team likes better. Thanks for your help!

Reply:
I was working on this last night and came to a solution that incorporates, but keeps the transforms in `pkg/config`. We add a transform function that runs on a hook that is called with `Load()`. This way we can parse the config during initial load, and other users can still use the config API. Please let me know what you think!